www/_blog/2022-03-29-graphql-now-available.mdx

```diff
@@ -0,0 +1,317 @@
+```
+
+Translates to the GraphQL base type
+
+```graphql
+type Account {
+  id: Int!
+  email: String!
+  createdAt: DateTime!
+  updatedAt: DateTime!
+}
+```
+
+And exposes bulk CRUD operations on the `Query` and `Mutation` types, complete with relay style keyset pagination, filters, and ordering and (optional) name inflection.
+
+```graphql
+type Query {
+  accountCollection(
+    first: Int
+    last: Int
+    before: Cursor
+    after: Cursor
+    filter: AccountFilter
+    orderBy: [AccountOrderBy!]
+  ): AccountConnection
+}
+
+type Mutation {
+	insertIntoAccountCollection(
+		objects: [AccountInsertInput!]!
+	): AccountInsertResponse
+
+	updateAccountCollection(
+    set: AccountUpdateInput!
+    filter: AccountFilter
+    atMost: Int! = 1
+  ): AccountUpdateResponse!
+
+  deleteFromAccountCollection(
+   filter: AccountFilter
+    atMost: Int! = 1
+  ): AccountDeleteResponse!
+```
+
+For a complete example with relationships, check out the [reflection docs](https://supabase.github.io/pg_graphql/reflection/).
+
+## Security
+
+An advantage to embedding GraphQL directly in the database is that we can lean on PostgreSQL's built-in primitives for authentication and authorization.
+
+### Authentication
+
+The GraphQL types exposed by `pg_graphql` are filtered according to the SQL role's [INSERT/UPDATE/DELETE permissions](https://www.postgresql.org/docs/current/sql-grant.html). 
+At Supabase, each API request is resolved in the database using the role in the request's JWT. 
+
+Anonymous users receive the `anon` role, and logged in users get the `authenticated` role. In either case, pg_graphql resolves requests according to the SQL permissions. 
+The [introspection schema](https://graphql.org/learn/introspection/) is similarly filtered to limit exposed types and fields to those that the user has permission to access. 
+That means we can serve multiple GraphQL schemas for users of differing privilege levels from a single endpoint!
+
+### Authorization
+
+Another nice side effect of making PostgreSQL do the heavy lifting is that GraphQL queries respect your existing 
+[row level security policies](https://www.postgresql.org/docs/current/ddl-rowsecurity.html) right out-of-the-box. No additional configuration required.
+
+## Performance
+
+Each [free tier database](https://supabase.com/pricing) on the Supabase platform runs on a dedicated AWS t4g.micro instance with 2 vCPUs and 1 GB of memory. 
+To squeeze the most out of that limited hardware we had to make a few significant optimizations:
+
+**GraphQL queries are always transpiled into exactly one SQL query**
+
+The SQL queries select and aggregate requested data into the shape of the GraphQL JSON response. 
+In addition to solving the [N+1 query problem](https://medium.com/the-marcy-lab-school/what-is-the-n-1-problem-in-graphql-dd4921cb3c1a), a common issue with GraphQL resolvers, 
+GraphQL queries requiring multiple joins typically produce significantly less IO due to reduced data duplication.
```

Comment:
is this correct @Oliver Rice? In the example below I assume that the pg_graphql is still doing the same amount of IO, it’s just passing the data back in a “deduplicated” format